frontend/src/components/ResourceNameTooltip.tsx

```diff
@@ -9,52 +9,87 @@ import {
-                      {resource.metadata.name}
-                    </DescriptionListDescription>
-                  </DescriptionListGroup>
-                  <DescriptionListGroup>
-                    <DescriptionListTerm>Resource type</DescriptionListTerm>
-                    <DescriptionListDescription>{resource.kind}</DescriptionListDescription>
-                  </DescriptionListGroup>
-                </DescriptionList>
-              </StackItem>
-            </Stack>
-          }
-        >
-          <Icon isInline aria-label="More info" role="button" tabIndex={0}>
-            <OutlinedQuestionCircleIcon />
-          </Icon>
-        </Tooltip>
-      </div>
-    )}
-  </div>
-);
+const ResourceNameTooltip: React.FC<ResourceNameTooltipProps> = ({
+  children,
+  resource,
+  resourceName,
+  resourceKind,
+}) => {
+  const [isHovered, setIsHovered] = React.useState(false);
+
+  const copyToClipboard = async (text: string) => {
+    navigator.clipboard.writeText(text);
+  };
+
+  const name = resourceName ?? resource.metadata?.name;
+  const kind = resourceKind ?? resource.kind;
+
+  return (
+    <div>
+      {children}{' '}
+      {name && (
+        <div style={{ display: 'inline-block' }}>
+          <Tooltip
+            removeFindDomNode
+            position="right"
+            content={
+              <Stack hasGutter>
+                <StackItem>
+                  Resource names and types are used to find your resources in OpenShift.
+                </StackItem>
+                <StackItem>
+                  <DescriptionList isCompact isHorizontal>
+                    <DescriptionListGroup>
+                      <DescriptionListTerm>Resource name</DescriptionListTerm>
+                      <DescriptionListDescription>{name}</DescriptionListDescription>
+                    </DescriptionListGroup>
+                    <DescriptionListGroup>
+                      <DescriptionListTerm>Resource type</DescriptionListTerm>
+                      <DescriptionListDescription>{kind}</DescriptionListDescription>
+                    </DescriptionListGroup>
+                  </DescriptionList>
+                </StackItem>
+                <StackItem>
+                  <Icon isInline>
+                    <InfoAltIcon />
+                  </Icon>{' '}
+                  <i>Click to copy resource name</i>
+                </StackItem>
+              </Stack>
+            }
+          >
+            <Icon
+              onMouseEnter={() => setIsHovered(true)}
+              onMouseLeave={() => setIsHovered(false)}
+              onClick={() => {
+                if (name) {
+                  copyToClipboard(name);
+                }
+              }}
+              isInline
+              style={{ cursor: 'pointer' }}
+              aria-label="More info"
+              tabIndex={0}
+            >
+              {isHovered ? <CopyIcon /> : <OutlinedQuestionCircleIcon />}
+            </Icon>
```

Comment:
I think as @DaoDaoNoCode mentioned on slack... you guys will have to work out this, [conflicts](https://github.com/opendatahub-io/odh-dashboard/pull/1529/commits/eaf0e2201672b465b8b0324bccdd693aec5fa247) be afoot... are the actions needed on the `Icon`?

Reply:
we ended up changing the design in a call with @vconzola and @DaoDaoNoCode in order to better align with both PRs and a UX aligned with Patternfly design guidelines
<img width="384" alt="Screenshot 2023-07-21 at 7 55 05 AM" src="https://github.com/opendatahub-io/odh-dashboard/assets/12587674/5ed13fb9-66ce-46e5-bbd5-d3ea6ca93eba">

https://github.com/opendatahub-io/odh-dashboard/assets/12587674/fa7c26e2-9e69-4bdc-a082-f8d5415e91b5

Reply:
ok so this is as a popover
<img width="441" alt="Screenshot 2023-07-21 at 8 29 51 AM" src="https://github.com/opendatahub-io/odh-dashboard/assets/12587674/e7df5961-0f42-4ac7-8070-0b77911c8ce6">

I can always revert back to the original idea